lib/logger/lib/logger/backends/console.ex

```diff
@@ -72,15 +109,44 @@ defmodule Logger.Backends.Console do
       enabled: Keyword.get(colors, :enabled, IO.ANSI.enabled?)}
   end
 
-  defp log_event(level, msg, ts, md, %{colors: colors, device: device} = state) do
-    output =
-      format_event(level, msg, ts, md, state)
-      |> color_event(level, colors)
-    send(device, {:io_request, self(), self(), {:put_chars, :unicode, output}})
+  defp log_event(level, msg, ts, md, %{device: device} = state) do
+    output = format_event(level, msg, ts, md, state)
+    %{state | ref: async_io(device, output), last_ts: ts}
+  end
+
+  defp buffer_event(level, msg, ts, md, state) do
+    %{buffer: buffer, buffer_size: buffer_size} = state
+    buffer = [buffer | format_event(level, msg, ts, md, state)]
+    %{state | buffer: buffer, buffer_size: buffer_size + 1, buffer_ts: ts}
+  end
+
+  defp async_io(device, output) do
+    pid = GenServer.whereis(device)
```

Comment:
This is not technically the right call, though we never get a device that isn't `:user` or `pid`. We should consider special casing `:user` to be more explicit about what is going on:
```elixir
defp async_io(:user, output) do
  case Process.whereis(:user) do
    pid when is_pid(pid) -> async_io(pid, output)
    nil                  -> raise "no device registered with name :user"
  end
end
```